Makefile

```diff
@@ -0,0 +1,84 @@
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: manager
+
+# Run tests
+test: generate fmt vet manifests
+	go test ./... -coverprofile cover.out
+
+# Build manager binary
+manager: generate fmt vet
+	go build -o bin/manager main.go
+
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: generate fmt vet manifests
+	go run ./main.go
+
+# Install CRDs into a cluster
+install: manifests
+	kustomize build config/crd | kubectl apply -f -
+
+# Uninstall CRDs from a cluster
+uninstall: manifests
+	kustomize build config/crd | kubectl delete -f -
+
+# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: manifests
+	cd config/manager && kustomize edit set image controller=${IMG}
+	kustomize build config/default | kubectl apply -f -
```

Comment:
For all the reading through this file to figure out what it does and doesn't do in `deploy` vs `install` vs other commands, maybe they should just read the readme? This way the onus is on them if they mess up the commands. And this way they learn.

Reply:
We can certainly revisit this. Do note that the typical user will not run the make file to deploy to their cluster.  I see this make file for dev purposes only. I think the user will just use the Kustomize files directly (we can bundle these files and attach them to the github release 😉 ) or do something like this:


$ kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v0.15.0/cert-manager.yaml

//IMHO I think that's the way to go. We just put that in our README

Reply:
Sounds good. I can't imagine why we'd want to skip validation though, seems like it's asking for trouble?

Reply:
BTW, I was meaning the developers when I said user.

Reply:
yeah. I just copy/pasted the command from cert-manager.io. Not sure why they have it that way. 🤷‍♂️